bin/sync_vsc_email_postfix.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python
+#
+# Copyright 2013-2019 Ghent University
+#
+# This file is part of vsc-administration,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# the Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# https://github.com/hpcugent/vsc-administration
+#
+# All rights reserved.
+#
+"""
+This script creates the list of canonical email adresses for VSC users.
+"""
+from __future__ import print_function
+
+import logging
+import sys
+
+from vsc.accountpage.client import AccountpageClient
+from vsc.accountpage.sync import Sync
+from vsc.accountpage.wrappers import mkAccount
+
+
+class VscPostfixSync(Sync):
+    CLI_OPTIONS = {
+        postfix_address_book: ('Location of the postfix canonical address file', None, 'store', None),
+        token: ('OAuth token to access the account page', None, 'store', None)
+    }
+
+    def do(self, dryrun):
+        """Actual work.
+
+        - retrieve all the email addresss from changed users
+        - read the current file
+        - add new/changed email addresses if needed
+        - write file
+        """
+        client = AccountpageClient(token=self.options.token)
+        active_accounts, inactive_accounts = self.get_accounts()
+
+        active_emails = dict([(a.vsc_id, a.email) for a in active_accounts])
+
+        with open(self.options.postfix_address_book, 'r') as pab:
+            address_map = dict( [ (v, t) for [v, t] in [l.split() for l in pab.readlines()] ])
+
+        target_emails = address_map.update(active_emails)
+
+        with open(self.options.postfix_address_book, 'w') as pab:
+            pab.write("\n".join(["%s    %s" % (k, v) for (k, v) in target_emails]))
```

Comment:
add trailing newline
`"%s %s" % x for x in target_emails` works too
what's up with all the spaces?